src/invocation/InvocationService.ts

```diff
@@ -24,13 +25,16 @@ class InvocationService {
         return deferred.promise;
     }
 
-    public invokeOnPartition(clientMessage: ClientMessage, partitionId: number): Q.Promise<ClientMessage> {
+    invokeOnPartition(clientMessage: ClientMessage, partitionId: number): Q.Promise<ClientMessage> {
         clientMessage.setPartitionId(partitionId);
-        var connection: ClientConnection = this.client.getConnectionManager().getOwnerConnection();
-        return this.invokeOnConnection(connection, clientMessage);
+        var address: Address = this.client.getPartitionService().getAddressForPartition(partitionId);
+        return this.client.getConnectionManager().getOrConnect(address)
```

Comment:
invokeOnPartition does not imply sending the request to specific member that holds the partition.I think in our terms `invokeOnPartition` means just setting partitionId accordingly (See https://github.com/hazelcast/hazelcast-python-client/blob/master/hazelcast/invocation.py#L139 ). 
Sending the request to a specific connection or owner connection should be determined later (See https://github.com/hazelcast/hazelcast-python-client/blob/master/hazelcast/invocation.py#L128 )
With these changes, we remove DummyClient option. Am I wrong?